src/hex_tarball.erl

```diff
@@ -43,8 +88,32 @@
 %%        inner_checksum => <<178,12,...>>}}
 %% '''
 %% @end
--spec create(metadata(), files()) -> {ok, {tarball(), checksum()}}.
+-spec create(metadata(), files()) -> {ok, tarball()} | {error, errors(), warnings()}.
```

Comment:
Adjusted the spec here and the tarball() type. We need to get dialyzer passing, which is not just due to this, but that should be done in another PR vs making this one bigger.

Reply:
I have adjusted this to function to return a map in both cases.... are we good with that? If not I can revert back to returning a tuple.